st2common/st2common/router.py

```diff
@@ -365,13 +365,15 @@ def __call__(self, req):
 
                 # NOTE: HACK: Workaround for eventlet wsgi server which sets Content-Type to
                 # text/plain if Content-Type is not provided in the request.
-                # All ouf our API endpoints except /exp/validation/mistral expect application/json
-                # so we explicitly set it to that if not provided (set to text/plain by the base
-                # http server) and if it's not /exp/validation/mistral API endpoint
+                # All ouf our API endpoints except /v1/workflows/inspection and
+                # /exp/validation/mistral expect application/json so we explicitly set it to that
+                # if not provided (set to text/plain by the base http server) and if it's not
+                # /v1/workflows/inspection and /exp/validation/mistral API endpoints.
                 if not self.is_gunicorn and content_type == 'text/plain':
```

Comment:
An alternative would be to update our code to support "arbitrary content types" (really just `application/json` for everything else and `text/plain` for this endpoint).

This would require us, at the very least, to update `openapi.yaml.j2` file and define expected / valid content type for each API controller and making necessary changes in the router.

This wouldn't be too much work and worth thinking about it while we are already working on this (so we can get rid of this ugly hack).

Reply:
I'm more incline to keep the hack as is. If this doesn't work, I probably would fix what you suggested and update all the API controllers. But since this hack already exists and i'm not deviating, I prefer to keep this change. If we want to improve the API in general, that seems like a separate PR to me.